### PR TITLE
Initial implementation of the up command (closes #1)

### LINKED ISSUE #1: Implement `up` command

### PR DESCRIPTION
**Pull request by jritsema**
This initial implementation of `up` operates on existing shipment(s) and does the following:
- updates replicas
- adds `BARGE` env var
- updates env vars
- supports multiple containers
- supports multiple shipments
- triggers

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,14 +6,14 @@ Using Harbor Compose is basically a four-step process.
 
 1. Define your app’s environment with a `Dockerfile` so it can be reproduced anywhere.
 
-2. Define the services that make up your app in `docker-compose.yml` so they can be run together in an isolated environment.  You can use the standard Docker Compose commands (like `docker-compose build` and `docker-compose up`) to build/run/test your Docker app locally.
+2. Define the services that make up your app in `docker-compose.yml` so they can be run together in an isolated environment.  You can use the standard docker compose commands (like `docker-compose build`, `docker-compose push`, `docker-compose up`, etc.) to build/run/test your Docker app locally.
```
Comment (wilsonodk) [inline README.md]: Why did you move away from using the proper noun for Docker Compose. In Docker's own Compose documentation, they use the proper noun when not referring to the actual commands. I think we should do the same. https://docs.docker.com/compose/overview/

**Comment by wilsonodk** [comment]
Are you planning one documenting the `harbor-compose.yml` similarly to the way Docker Compose is documented? [Docker Compose File Reference](https://docs.docker.com/compose/compose-file/)

File: README.md (hunk 1 above)
Comment (jritsema) [inline README.md]: Good point.  I'll change it back.

**Comment by jritsema** [comment]
Yes, I'll add an issue to document for the file format.

**Comment by jritsema** [comment]
https://github.com/turnerlabs/harbor-compose/issues/12

File: cmd/up.go [hunk 2]
```diff
@@ -19,5 +21,114 @@ func init() {
...
 
 func up(cmd *cobra.Command, args []string) {
-	fmt.Println("TODO")
+
+	//read the harbor compose file
+	harborCompose := DeserializeHarborCompose(HarborComposeFile)
+
+	//read the docker compose file
+	dockerCompose := DeserializeDockerCompose(DockerComposeFile)
+	if dockerCompose.Version != "2" {
```
Comment (jkurz) [inline cmd/up.go]: should this check be in the Deserialize function? That way we don't have to do it in every function.

File: cmd/harborAPI.go [hunk 3]
```diff
@@ -128,35 +149,56 @@ func update(token string, url string, data interface{}) {
...
-		uri, _ := template.Expand(values)
-
-		_, body, err := gorequest.New().
-			Get(uri).
-			End()
-
-		if err != nil {
-			log.Fatal(err)
-		}
+	values := make(map[string]interface{})
```
Comment (jkurz) [inline cmd/harborAPI.go:177]: looks like I need to get on the same page with you when it comes to formatting. This is go fmt doing this as a plugin in VSCode?

File: cmd/harborAPI.go [hunk 4]
```diff
@@ -170,13 +212,89 @@ func Trigger(shipment string, env string) {
...
+		values["container"] = container
+		templateString = shipItURI + "/v1/shipment/{shipment}/environment/{env}/container/{container}/envvar/{envvar}"
+	}
+
+	template, _ := uritemplates.Parse(templateString)
+	url, _ := template.Expand(values)
+
+	//issue delete call
+	//api will return 422 if the envvar doesn't exist, which can be ignored
+	res, _, _ := delete(token, url)
```
Comment (jkurz) [inline cmd/harborAPI.go:256]: why are we deleting the envars here?

Comment (jritsema) [inline cmd/harborAPI.go:256]: AFAIK there's no update API

Comment (jkurz) [inline cmd/harborAPI.go:256]: There are PUTs for every endpoint. https://bitbucket.org/vgtf/mss-shipit-api

Comment (jritsema) [inline cmd/harborAPI.go:256]: I thought I had tried those a while back and had issues.  Let's just defer this for now until we rewrite up.  I just want to get this early version out the door.
